flink-table-store-core/src/main/java/org/apache/flink/table/store/table/AbstractFileStoreTable.java

```diff
@@ -40,6 +47,43 @@ public AbstractFileStoreTable(Path path, TableSchema tableSchema) {
 
     protected abstract FileStore<?> store();
 
+    protected abstract FileStoreTable copy(TableSchema newTableSchema);
+
+    @Override
+    public FileStoreTable copy(Map<String, String> dynamicOptions) {
+        // check option is not immutable
+        Map<String, String> options = tableSchema.options();
+        dynamicOptions.forEach(
+                (k, v) -> {
+                    if (!Objects.equals(v, options.get(k))) {
```

Comment:
Should the immutable field be exist in the dynamic parameters? Maybe it's better to `SchemaManager.checkAlterTableOption(k)` without comparing the value

Reply:
The immutable field may be exist in the dynamic parameters.
For example, for Flink, Flink planner just pass a options merged table options and dynamic options.